framework/src/preference/private_preference.c

```diff
@@ -198,6 +217,29 @@ int preference_get_string(const char *key, char **value)
 	return ret;
 }
 
+int preference_get_binary(const char *key, void **value, int *len)
+{
+	int ret;
+	preference_data_t data;
+
+	if (key == NULL) {
+		return PREFERENCE_INVALID_PARAMETER;
+	}
+
+	data.key = key;
+	data.type = PRIVATE_PREFERENCE;
+	data.attr.type = PREFERENCE_TYPE_BINARY;
+
+	/* Get preference with prctl */
+	ret = prctl(PR_GET_PREFERENCE, &data);
+	if (ret == OK) {
+		*value = (void *)data.value;
```

Comment:
Don't you need to check `*value` is null?

Reply:
NULL returned values should be checked as general code practice, so I understand your point.
However, I have 2 considerations here: -
1. The writing of preference value is accompanied by checksum, so a successful write return is a good estimator of confirmation. Also, in `get_preference`, `read_fs_key` function involves CRC verification of the read preference. I think these 2 are enough for safeguarding us. Any additional logical checks of the key-value should be in the hands of the use-case developer.
2. No other `get_preference` API checks for NULL values. All should be updated together for consistency.

Hence, I feel current version is alright.

Reply:
Update: - while int, bool and double values are updated in the variable passed by reference and the space from the kernel side is freed,
In the case of string and binary preferences, only a pointer is passed from the application side to retrieve the address of the variable holding the preference data/value. It is initialized as NULL and updated with the address of the variable holding the preference data. In this case, the space from the kernel side is not freed and it is still pointed to by the pointer sent from the application call to preference API.
Hence I feel it is OK to initialize string and binary preference value pointers as NULL, and they should not be checked for NULL values, because it is intended.

the `len` pointer should however be checked for NULL values as it directly retrieves the value stored and does not simply point to it.

@kishore-sn @sunghan-chang

Reply:
Updated as per suggestions from yu and @kishore-sn